app/models/canonical/ingredient.rb

```diff
@@ -17,10 +22,39 @@ class Ingredient < ApplicationRecord
 
     validates :name, presence: true
     validates :item_code, presence: true, uniqueness: { message: 'must be unique' }
+    validates :ingredient_type, inclusion: { in: VALID_TYPES, message: TYPE_VALIDATION_MESSAGE, allow_blank: true }
     validates :unit_weight, presence: true, numericality: { greater_than_or_equal_to: 0 }
+    validates :purchase_requires_perk,
+              inclusion: {
+                           in:      BOOLEAN_VALUES,
+                           message: "#{BOOLEAN_VALIDATION_MESSAGE} if purchasable is true",
+                         },
+              if:        -> { purchasable == true }
+    validates :unique_item, inclusion: { in: BOOLEAN_VALUES, message: BOOLEAN_VALIDATION_MESSAGE }
```

Comment:
The fact `unique_item` has a default value prevents this from ever really coming up. If the value is `nil` it will be automatically set to `false` prior to validation, and truthy values will be set to `true` prior to validation whether there is a default value or not.